server/src/main/java/org/elasticsearch/action/admin/indices/settings/get/TransportGetSettingsAction.java

```diff
@@ -69,25 +73,39 @@ protected GetSettingsResponse newResponse() {
         return new GetSettingsResponse();
     }
 
+    private static boolean isFilteredRequest(GetSettingsRequest request) {
+        return CollectionUtils.isEmpty(request.names()) == false;
+    }
+
     @Override
     protected void masterOperation(GetSettingsRequest request, ClusterState state, ActionListener<GetSettingsResponse> listener) {
         Index[] concreteIndices = indexNameExpressionResolver.concreteIndices(state, request);
         ImmutableOpenMap.Builder<String, Settings> indexToSettingsBuilder = ImmutableOpenMap.builder();
+        ImmutableOpenMap.Builder<String, Settings> indexToDefaultSettingsBuilder = ImmutableOpenMap.builder();
         for (Index concreteIndex : concreteIndices) {
             IndexMetaData indexMetaData = state.getMetaData().index(concreteIndex);
             if (indexMetaData == null) {
                 continue;
             }
 
-            Settings settings = settingsFilter.filter(indexMetaData.getSettings());
+            Settings indexSettings = settingsFilter.filter(indexMetaData.getSettings());
             if (request.humanReadable()) {
-                settings = IndexMetaData.addHumanReadableSettings(settings);
+                indexSettings = IndexMetaData.addHumanReadableSettings(indexSettings);
             }
-            if (CollectionUtils.isEmpty(request.names()) == false) {
-                settings = settings.filter(k -> Regex.simpleMatch(request.names(), k));
+
+            if (isFilteredRequest(request)) {
+                indexSettings = indexSettings.filter(k -> Regex.simpleMatch(request.names(), k));
+            }
+
+            indexToSettingsBuilder.put(concreteIndex.getName(), indexSettings);
+            if (request.includeDefaults()) {
+                Settings defaultSettings = settingsFilter.filter(indexScopedSettings.diff(indexSettings, this.settings));
```

Comment:
This looks off; `this.settings` will never have index-scoped settings in it as these are node-level settings only.

Reply:
I think you want `Settings.EMPTY` here.

Reply:
good one, is this a bug that we already had given that we were doing the same in the REST action before? or did things work by coincidence?

Reply:
My suspicion is that this is leftover from when this `settings` instance could contain index-level settings (think back to 2.x when you could set `index.number_of_shards` at the node-level, for example). Today though, there are not any index-level settings that can be set at the node level, so there will never be a match for any setting that is at the index-level, which means this can safely be replaced by `Settings.EMPTY` and behave exactly as its behaving right now, except it will not "look" wrong.

Reply:
agreed, thanks for the explanation

Reply:
Addressed the settings filter issues @jasontedor